### PR TITLE
feat(Instructor): introduce Instructor Hub with tutorials, examples, and new CLI

### PR DESCRIPTION
**Pull request by jxnl**
| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit 4b62a35a6f84e7ae1551484003689c03e3e6b436.  | 
|--------|--------|

### Summary:
This PR introduces the 'Instructor Hub', a new CLI, updates documentation, adds tests, demonstrates the OpenAI API, adds a Cloudflare worker, modifies the `get_py` function, removes old tutorials, creates a new SQL table for analytics, updates request and response handling, removes the `mkdoc_yaml_url` function, and updates the `HubClient` class and `pull` function.

**Key points**:
- Introduced `Instructor Hub` with tutorials and examples.
- Implemented a new CLI for interaction with the hub.
- Updated `mkdocs.yml` to include the new hub in the documentation.
- Added tests for the new hub.
- Demonstrated single-label classification using the OpenAI API.
- Added a Cloudflare worker in `router.ts` for serving the hub content.
- Modified `get_py` function to include a source comment in the generated Python code.
- Removed old tutorials in the `tutorials` directory.
- Created a new SQL table `hub_analytics` for tracking analytics.
- Updated `index.ts` and `router.ts` in the `instructor-hub-proxy` directory to handle requests and responses, and to track analytics.
- Removed `mkdoc_yaml_url` function in `instructor/cli/hub.py`.
- Updated `HubClient` class in `instructor/cli/hub.py` to include methods for getting cookbooks by id and slug.
- Updated `pull` function in `instructor/cli/hub.py` to use these new methods.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: instructor/cli/hub.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    return f"https://raw.githubusercontent.com/jxnl/instructor/{branch}/mkdocs.yml?raw=true"
+
+
+def list_hub(branch="main") -> Iterable[HubPage]:
+    resp = httpx.get(mkdoc_yaml_url(branch))
+    mkdocs_config = resp.content.decode("utf-8").replace("!", "")
+    data = yaml.safe_load(mkdocs_config)
+
+    # Replace with Hub key
+    cookbooks = [obj["Hub"] for obj in data.get("nav", []) if "Hub" in obj][0]
```
[the remark this answers is not in the document]
Comment (jxnl) [reply instructor/cli/hub.py:65]: If they are missing we want an error

File: instructor/cli/hub.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    slug: Optional[str] = typer.Option(None, "--slug", "-s", help="The cookbook slug"),
+    py: bool = typer.Option(False, "--py", help="Output to a Python file"),
+    branch: str = typer.Option(
+        "hub", help="Specific branch to fetch the cookbooks from."
+    ),
+    page: bool = typer.Option(
+        False, "--page", "-p", help="Paginate the output with a less-like pager"
+    ),
+):
+    cookbook = (
```
[the remark this answers is not in the document]
Comment (jxnl) [reply instructor/cli/hub.py:130]: @hbrooks really reasonable!

File: instructor/cli/hub.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hbrooks) [reply instructor/cli/hub.py]: @jxnl, at the moment you'll need to explicitly tag us

**Comment by hbrooks** [comment]
Ellipsis is working on hard on this PR!

**Pull request merged**